src/web/arr/trove/chart.arr

```diff
@@ -122,6 +286,368 @@ y-max-method = method(self, y-max :: Number):
+
+  fun get-values(row :: RawArray) -> List<Number>: 
+    doc:```
+        VALUE GETTER: Gets the values from the row of data in List form
+        ASSUMES the row of data is ordered by [LABEL, VALUES, OTHER]
+        ```
+    raw-array-get(row, 1) ^ raw-array-to-list
+  end
+  
+  sum = {(l :: List<Number>): fold({(acc, elm): acc + elm}, 0, l)}
+  sorted-table = table-sorter(self.obj!tab, get-values, sum, cmp, eq)
+  self.constr()(self.obj.{tab: sorted-table})
+end
+
+axis-pointer-method = method(self,
+    tickValues :: List<Number>, 
+    tickLabels :: List<String>) block: 
+
+  # Lengths of Lists
+  TVLen = tickValues.length() 
+  TLLen = tickLabels.length()
+  distinctTVLen = distinct(tickValues).length()
+
+  # Edge Case Error Checking
+  when not(distinctTVLen == TVLen): 
+    raise('add-pointers: pointers cannot overlap')
+  end
+  when not(TVLen == TLLen): 
+    raise('add-pointers: pointers values and names should have the same length')
+  end
+
+  ticks = fold2({(acc, e1, e2): link(pointer(e1, e2), acc)}, empty, tickLabels, tickValues)
+  self.constr()(self.obj.{pointers: some(distinct(ticks) ^ builtins.raw-array-from-list)})
+end
+
+make-axis-data-method = method(self,  pos-bar-height :: Number, neg-bar-height :: Number):
+  step-types = [list: 0, 0.2, 0.25, 0.5, 1, 2]
+
+  # Turn the numbers into Scientific Numbers
+  scientific-b10 = num-to-scientific(10)
+  pos-sci = scientific-b10(pos-bar-height)
+  neg-sci = scientific-b10(neg-bar-height)
+
+  # Calculate the step distance between gridlines
+  pos-step = step-types.filter({(n): n >= num-abs(pos-sci.coeff / 9)}).get(0) * num-expt(10, pos-sci.exponent)
+  neg-step = step-types.filter({(n): n >= num-abs(neg-sci.coeff / 9)}).get(0) * num-expt(10, neg-sci.exponent)
+  step = num-max(pos-step, neg-step)
+  step-sci = scientific-b10(step)
+
+  # Use step distance to calculate Axis Properties
+  name-tick = 
+    {(n): 
+      ask:
+      | (step-sci.coeff == 2.5) and (step-sci.exponent <= 0) then: 
+        pointer(num-to-string-digits(n, 2 - step-sci.exponent), n)
+      | step-sci.exponent < 0 then: 
+        pointer(num-to-string-digits(n, 1 - step-sci.exponent), n)
+      | otherwise: 
+        pointer(num-to-string(n), n)
+      end}
+
+  axisTop = num-max(0, step * num-ceiling(pos-bar-height / step))
+  axisBottom = num-min(0, step * num-floor(neg-bar-height / step))
+  pos-ticks = map(name-tick, range-by(0, axisTop + step, step))
+  neg-ticks = map(name-tick, range-by(0, axisBottom - step, -1 * step))
+  ticks = distinct(pos-ticks + neg-ticks) ^ builtins.raw-array-from-list
+  self.constr()(
+    self.obj.{axisdata: some(axis-data(axisTop, axisBottom, ticks))}
+    )
+end
+
+format-axis-data-method = method(self, format-func :: (Number -> String)):
+  cases (Option) self.obj!axisdata: 
+    | none => 
+      raise("Should never have reached this point. Yell at John for not setting up the axis properties somewhere where he should have and please report this as a bug")
```

Comment:
Clean up this error message...

Reply:
Yeah really sorry about this, I meant to go back and change it, Doing that now!